app/models/shipit/deploy_spec/yarn_discovery.rb

```diff
@@ -0,0 +1,49 @@
+module Shipit
+  class DeploySpec
+    module YarnDiscovery
+      def discover_dependencies_steps
+        discover_yarn || super
+      end
+
+      def discover_yarn
+        yarn_install if yarn?
+      end
+
+      def yarn_install
+        [%(yarn install --no-progress)]
+      end
+
+      def discover_review_checklist
+        discover_yarn_checklist || super
+      end
+
+      def discover_yarn_checklist
+        [%(<strong>Don't forget version and tag before publishing!</strong> You can do this with:<br/>
+          yarn version --new-version <strong>&lt;major|minor|patch&gt;</strong> && git push --tags</pre>)] if yarn?
+      end
+
+      def yarn?
+        yarn_lock.exist? && public?
```

Comment:
¯\\_(ツ)_/¯ I liked that yarn's precedence over npm was made clear in `FileSystem`'s include order, and -- while it might not be obvious -- the tests are harder to reason about now.

Regardless, I've updated to a more concise implementation (a more greppable/verbose interim implmentation is at https://github.com/Shopify/shipit-engine/pull/638/commits/5a728f8d583fd587f371939a02bc51fef2a491bf).